tooling/internal/matlab/static_files/+binary/FixedVectorSerializer.m

```diff
@@ -0,0 +1,51 @@
+% Copyright (c) Microsoft Corporation.
+% Licensed under the MIT License.
+
+classdef FixedVectorSerializer < yardl.binary.VectorSerializerBase
+    properties
+        length_;
+    end
+
+    methods (Access=protected)
+        function handle_write_count_(self, ~, count)
+            if count ~= self.length_
+                throw(yardl.ValueError("Expected an array of length %d, got %d", self.length_, count));
```

Comment:
A future improvement would be to provide a path with these errors, so the user has a better idea of what value caused the error. Applies to Python as well.

Reply:
Strongly agree, especially for records with many and/or nested fields. I vote to open a feature request issue after this PR.